cpp/ql/src/Diagnostics/FailedCompilations.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Failed compiler invocations.
+ * @description Gives the command-line of compilations for which extraction did not run to completion.
+ * @kind diagnostic
+ * @id cpp/diagnostics/failed-compilations
+ */
+
+import cpp
+
+class AnonymousCompilation extends Compilation {
+  override string toString() { result = "<compilation>" }
+}
+
+string describe(Compilation c) {
+  if c.getArgument(1) = "--mimic"
+  then result = "compiler invocation " + concat(int i | i > 1 | c.getArgument(i), " " order by i)
```

Comment:
Why should we report compiler invocations with errors? That could be deliberate, like when running `configure`. If it's not a recoverable error, the build as a whole hopefully failed.

Reply:
That's code from @igfoo, can you elaborate on that?

Reply:
As opposed to "extractor invocation" errors, you mean? The full code is:
```
  if c.getArgument(1) = "--mimic"
  then result = "compiler invocation " + concat(int i | i > 1 | c.getArgument(i), " " order by i)
  else result = "extractor invocation " + concat(int i | | c.getArgument(i), " " order by i)
```
What's happening here is normally in a build we'd expect the extractor to be called with something like
```
$extractor --mimic /usr/bin/gcc ...
```
in which case this reports a failure for the `compiler invocation gcc ...` (although it's the extractor that failed). But it is possible that something is invoking the extractor differently, in which case we don't know what compiler invocation it corresponds to, if any. So they are really both extractor invocations, but I used different leading text so that we could tell what is happening.

It is true that we will run the extractor even if the compiler failed, in which case we'd report a failure here even when the failure is expected. We don't currently have the information necessary to report that differently, though.

Reply:
Thanks for the clarification! It looks like this code is what we want for now.